From: Shengjing Zhu <zhsj@debian.org>
Date: Fri, 5 Feb 2021 21:21:45 +0800
Subject: skip userns test in schroot as well

When schroot is using overlayfs, it fails to detect it as chroot.
---
 src/syscall/exec_linux_test.go | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/syscall/exec_linux_test.go b/src/syscall/exec_linux_test.go
index a6900f9..221f9de 100644
--- a/src/syscall/exec_linux_test.go
+++ b/src/syscall/exec_linux_test.go
@@ -35,6 +35,10 @@ func isLXC() bool {
 	return os.Getenv("container") == "lxc"
 }
 
+func isSchroot() bool {
+	return os.Getenv("SCHROOT_SESSION_ID") != ""
+}
+
 func skipInContainer(t *testing.T) {
 	// TODO: the callers of this func are using this func to skip
 	// tests when running as some sort of "fake root" that's uid 0
@@ -50,6 +54,9 @@ func skipInContainer(t *testing.T) {
 	if isLXC() {
 		t.Skip("skip this test in LXC container")
 	}
+	if isSchroot() {
+		t.Skip("skip this test in Schroot container")
+	}
 }
 
 func skipNoUserNamespaces(t *testing.T) {
